### PR TITLE
Use atomic pushes for better error handling

### PR DESCRIPTION
**Pull request by LinusU**
Atomic pushes will fail if all parts can't be pushed. What this means in this case is that if anyone else has pushed something to master while you where trying to do your release, it will gracefully fail.

**Limited testing**, might be good to test it our yourself first 💃

### CODE REVIEW

**Comment by pinn3** [comment]
Will do tonight 👍

File: index.js [hunk 1]
```diff
@@ -15,18 +17,24 @@ const releasePackage = () => {
...
+    return execa('npm', ['publish'])
+  }).then(() => {
+    return tag
   })
 }
 
 if (require.main === module) {
-  releasePackage()
+  releasePackage().then(
+    tag => console.log(`🚀  ${tag}`),
```
Comment (pinn3) [inline index.js]: 💨 🚀

**Comment by pinn3** [comment]
Did some testing on https://github.com/pinn3/4a0d-11e6 and noticed some pretty major issues, mostly my fault;

1. 🙈  I forgot to add the bin field to the `package.json`, causing the package to unusable in cli execution.
2. 🙈  Also forgot to add the environment header to `index.js`
3. 😢  The graceful fallback still updates the package version, causing the next version bump to be on top of that.

Otherwise, very nice!
Will fix those issues that are on my end, could you take a look on the last one?

**Comment by LinusU** [comment]
I will fix 👌

**Comment by LinusU** [comment]
Done 🙌

File: index.js [hunk 2]
```diff
@@ -17,18 +20,34 @@ const releasePackage = () => {
...
+
+  yield execa('git', ['checkout', '-b', branchName])
+
+  try {
+    const tag = (yield execa('npm', ['version', releaseVersion])).stdout
+
+    try {
+      yield execa('git', ['push', '--atomic', 'origin', `${branchName}:master`, `refs/tags/${tag}`])
+    } catch (err) {
+      yield execa('git', ['tags', '-d', tag])
```
Comment (pinn3) [inline index.js]: `git tags` is not a valid command, should be `git tag`!

File: index.js [hunk 3]
```diff
@@ -17,18 +20,34 @@ const releasePackage = () => {
...
+      throw err
+    }
+
+    yield execa('npm', ['publish'])
+
+    return tag
+  } finally {
+    yield execa('git', ['checkout', 'master'])
+    yield execa('git', ['branch', '-D', branchName])
+  }
```
Comment (pinn3) [inline index.js]: Also maybe add a `git pull` so that the release doesn't leave you one commit behind master?

Comment (LinusU) [inline index.js]: :+1:

**Comment by pinn3** [comment]
Fixing these things myself

**Comment by LinusU** [comment]
Sorry about the delay

**Comment by pinn3** [comment]
No worries!

**Pull request merged**